src/NuGet.Clients/NuGet.SolutionRestoreManager/Directory.Build.targets

```diff
@@ -0,0 +1,4 @@
+<Project>
+    <Import Project="$([MSBuild]::GetPathOfFileAbove('Directory.Build.targets', '$(MSBuildThisFileDirectory)../'))" />
+    <Import Project="$(PkgMicrosoft_VSSDK_BuildTools)\tools\VSSDK\Microsoft.VsSDK.targets" Condition="'$(PkgMicrosoft_VSSDK_BuildTools)' != ''" />
```

Comment:
The `Microsoft.VsSDK.targets` file needs to be imported AFTER all the .NET/msbuild common targets are imported. Since the project file uses `<Project Sdk="Microsoft.NET.Sdk>`, the project file no longer explicitly imports the targets in the file, allowing us to add extra MSBuild script after the targets import.

However, [MSBuild allows you to explicitly import the SDK's props and targets files in the project file](https://learn.microsoft.com/en-us/visualstudio/msbuild/how-to-use-project-sdk?view=vs-2022). If we change to this syntax, we shouldn't need to add these `Directory.Build.targets` files.  What do people think?

For what it's worth, I also tried moving this import to `build/common.targets`, but for reasons I don't understand, it caused two resources files in the `NuGet.Tools` project to have the same output path, which failed the build.

Reply:
I think this more elegant, I'm hoping they fix it in the future too so its even easier to have and SDK-style project for VSIX.

Here's the alternative in action for others to compare: https://github.com/microsoft/slngen/blob/main/src/Microsoft.VisualStudio.SlnGen.Extension/Microsoft.VisualStudio.SlnGen.Extension.csproj#L49-L50